File: frontend/src/types/case.ts
// src/types/case.ts

/**
 * กำหนดประเภทของความสำคัญของคดี
 */
export type CasePriority = 'critical' | 'high' | 'medium' | 'low';

/**
 * กำหนดประเภทของสถานะคดี
 */
export type CaseStatus = 'open' | 'investigating' | 'closed' | 'suspended';

/**
 * กำหนดประเภทของหมวดหมู่คดี
 */
export type CaseType = 'cyber_crimes' | 'financial_crimes' | 'gambling' | 'fraud';

/**
 * โครงสร้างข้อมูลของคดี (Case Object)
 */
export interface Case {
  evidence?: Evidence[];
  case_id: string;
  title: string;
  description: string;
  case_type: CaseType;
  priority: CasePriority;
  status: CaseStatus;
  createdDate: string;   // รูปแบบ 'YYYY-MM-DD'
  lastUpdated: string;   // รูปแบบ 'YYYY-MM-DD'
  assignedOfficer: string;
  evidenceCount: number;
}

export interface newCase {
  title: string;
  description: string;
  case_type: CaseType;
  priority: CasePriority;
}

export interface Evidence {
  id: string;
  case_id: string; // ID ของคดีที่หลักฐานนี้เกี่ยวข้อง
  file_name: string; // ชื่อหลักฐาน
  url: string;
  description?: string; // อาจมีคำอธิบายเพิ่มเติมเกี่ยวกับหลักฐาน
  uploadedAt: string; // วันที่อัพโหลดหลักฐาน รูปแบบ 'YYYY-MM-DD'
}

File: backend/backend/models/case.py
from datetime import date
from enum import Enum
from sqlmodel import Field, SQLModel
from typing import Optional

class CaseType(str, Enum):
    CYBER = "cyber_crimes"
    FINANCE = "financial_crimes"
    CASINO = "casino_crimes"
    SCAM = "scam_crimes"
    ETC = "etc_crimes"

class CasePriority(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"

class CaseStatus(str, Enum):
    OPEN = "open"
    INVESTIGATING = "investigating"
    CLOSED = "closed"
    SUSPENDED = "suspended"

class Case(SQLModel, table=True):
    __tablename__ = "cases"

    id: Optional[int] = Field(default=None, primary_key=True)
    title: str = Field(index=True)
    case_type: CaseType
    description: Optional[str] = Field(default=None)
    priority: CasePriority
    status: CaseStatus
    created_date: date = Field(default=date.today())
    last_updated: date = Field(default=date.today())
    assigned_officer: str = Field(index=True)
    evidence_count: Optional[int] = Field(default=0)